### PR TITLE
Mark sensors as diagnostic

### PR DESCRIPTION
**Pull request by rfvermut**
Unclutters default lovelace view

image

### CODE REVIEW

**Comment by DurgNomis-drol** [comment]
Please make sure that all checks succeed. Run `poetry run pre-commit run --all-files` before commiting or to fix the errors that exists now and the push a new commit with these fixes 😊

**Comment by rfvermut** [comment]
Oh, it opened a can of worms:

```
  And because homeassistant (2021.11.5) depends on httpx (0.19.0)
   and mytoyota (0.7.4) depends on httpx (>=0.18.1,<0.19.0), mytoyota (0.7.4) is incompatible with homeassistant (>=2021.11.1,<2022.0.0).
  So, because toyota depends on both homeassistant (^2021.11.1) and mytoyota (0.7.4), version solving failed.
```

**Comment by rfvermut** [comment]
So depends on https://github.com/DurgNomis-drol/mytoyota/pull/105

File: custom_components/toyota/binary_sensor.py [hunk 1]
```diff
@@ -266,6 +271,7 @@ class ToyotaOverAllStatusBinarySensor(ToyotaBaseEntity, BinarySensorEntity):
     """Class for the overall warning sensor"""
 
     _attr_device_class = DEVICE_CLASS_PROBLEM
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
```
Comment (DurgNomis-drol) [inline custom_components/toyota/binary_sensor.py:274]: This should not be listed as diagnostic. It is meant for people to easily see if something is not right with car.

File: custom_components/toyota/binary_sensor.py [hunk 2]
```diff
@@ -320,6 +327,7 @@ class ToyotaFrontDefoggerSensor(ToyotaBaseEntity, BinarySensorEntity):
     """Class for HVAC sensor"""
 
     _attr_icon = ICON_FRONT_DEFOGGER
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
```
Comment (DurgNomis-drol) [inline custom_components/toyota/binary_sensor.py:330]: Part of the HVAC system and does not belong in the diagnostic category. This sensor is not in it's final for yet. as I am not sure what the different numbers means.

File: custom_components/toyota/binary_sensor.py [hunk 3]
```diff
@@ -332,6 +340,7 @@ class ToyotaRearDefoggerSensor(ToyotaBaseEntity, BinarySensorEntity):
     """Class for HVAC sensor"""
 
     _attr_icon = ICON_REAR_DEFOGGER
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
```
Comment (DurgNomis-drol) [inline custom_components/toyota/binary_sensor.py:343]: See above comment on FrontDefogger

File: custom_components/toyota/entity.py [hunk 4]
```diff
@@ -69,6 +70,7 @@ class StatisticsBaseEntity(ToyotaBaseEntity, SensorEntity):
     """Builds on Toyota base entity"""
 
     _attr_icon = ICON_HISTORY
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
```
Comment (DurgNomis-drol) [inline custom_components/toyota/entity.py:73]: Statistics does not belong in the diagnostic category.

**Review by DurgNomis-drol — CHANGES_REQUESTED** [review]
The `entity_category` is really not meant for integrations like this, though I like the initiative (Both yours and HA). Some sensors are straight forward, others like `starter_battery_health` would normally be in the `diagnostic` category, but in this instance it does not make sense. The same goes for some of those I made comments on below. As this is the first time we see this in HA, it is probably not in it's final form yet. This usually takes a couple of versions, so we will probably need to revisit this again.

**Comment by DurgNomis-drol** [comment]
Please rebase onto master, underlying issue with `httpx` has been fixed.

File: pyproject.toml [hunk 5]
```diff
@@ -7,7 +7,7 @@ license = "MIT"
 
 [tool.poetry.dependencies]
 python = "^3.8"
-homeassistant = "^2021.3.3"
+homeassistant = "^2021.11.1"
```
Comment (DurgNomis-drol) [inline pyproject.toml:10]:
```suggestion
homeassistant = ">=2021.11"
```

**Review by DurgNomis-drol — CHANGES_REQUESTED** [review]
One last thing! 😃

**Review by DurgNomis-drol — APPROVED** [review]
Awesome! Thanks for contributing! ❤️

**Pull request merged**